src/transaction_signature_options/sign_with_mpc/mod.rs

```diff
@@ -0,0 +1,529 @@
+use color_eyre::{eyre::Context, owo_colors::OwoColorize};
+use inquire::CustomType;
+use near_primitives::transaction::{Transaction, TransactionV0};
+
+use crate::common::{JsonRpcClientExt, RpcQueryResponseExt};
+
+mod mpc_sign_request;
+mod mpc_sign_result;
+mod mpc_sign_with;
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = crate::commands::TransactionContext)]
+#[interactive_clap(output_context = SignMpcContext)]
+pub struct SignMpc {
+    #[interactive_clap(skip_default_input_arg)]
+    /// What is the Admin account address?
+    admin_account_id: crate::types::account_id::AccountId,
+
+    #[interactive_clap(subargs)]
+    /// What is key type for deriving key?
+    mpc_key_type: MpcKeyType,
+}
+
+#[derive(Clone)]
+pub struct SignMpcContext {
+    admin_account_id: near_primitives::types::AccountId,
+    tx_context: crate::commands::TransactionContext,
+}
+
+impl SignMpcContext {
+    pub fn from_previous_context(
+        previous_context: crate::commands::TransactionContext,
+        scope: &<SignMpc as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        let _ = previous_context
+            .network_config
+            .get_mpc_contract_account_id()?;
+
+        Ok(SignMpcContext {
+            admin_account_id: scope.admin_account_id.clone().into(),
+            tx_context: previous_context,
+        })
+    }
+}
+
+impl SignMpc {
+    pub fn input_admin_account_id(
+        context: &crate::commands::TransactionContext,
+    ) -> color_eyre::eyre::Result<Option<crate::types::account_id::AccountId>> {
+        crate::common::input_signer_account_id_from_used_account_list(
+            &context.global_context.config.credentials_home_dir,
+            "What is the Admin AccountId?",
+        )
+    }
+}
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = SignMpcContext)]
+#[interactive_clap(output_context = MpcKeyTypeContext)]
+pub struct MpcKeyType {
+    #[interactive_clap(skip_default_input_arg)]
+    /// What is the MPC key type for derivation?
+    key_type: crate::types::key_type::KeyType,
+
+    #[interactive_clap(named_arg)]
+    /// What is the derivation path?
+    derivation_path: MpcDeriveKey,
+}
```

Comment:
There is a much simpler way:

1. use `subcommand` at the parent type (`SignMpc`) instead of `subargs` (here is a relevant example: https://github.com/near/near-cli-rs/blob/c9190cb677bb2c8c060aed698264d330d60c8edc/src/commands/contract/view_storage/keys_to_view/all_keys.rs#L6)
2. Turn this struct into enum:

```rust
enum MpcKeyType {
    Secp256K1(DerivationPath),
    Ed25519(DerivationPath),
}
```
3. Remove all the custom code that handles the key type selection